nuclio_sdk/json_encoder.py

```diff
@@ -13,10 +13,72 @@
 # limitations under the License.
 
 import json
+import json.encoder
 
 
 # JSON encoder that can encode custom stuff
 class Encoder(json.JSONEncoder):
+    def __init__(self, **kwargs):
+        super(Encoder, self).__init__(**kwargs)
+        self.nan_str = "NaN"
+
+    def iterencode(self, o, _one_shot=False):
+        """Encode the given object and yield each string
+        representation as available.
+
+        For example::
+
+            for chunk in JSONEncoder().iterencode(bigobject):
+                mysocket.write(chunk)
+
+        """
+        if self.check_circular:
+            markers = {}
+        else:
+            markers = None
+        if self.ensure_ascii:
+            _encoder = json.encoder.encode_basestring_ascii
+        else:
+            _encoder = json.encoder.encode_basestring
+
+        def floatstr(
+            o,
+            allow_nan=self.allow_nan,
+            nan_str=self.nan_str,
+            _repr=float.__repr__,
+            _inf=json.encoder.INFINITY,
+            _neginf=-json.encoder.INFINITY,
+        ):
+            if o != o:
```

Comment:
type check?

Reply:
will use `math.isnan` for readability but do note that this is the usual way (from stdlib) to check whether float object is NaN

Reply:
Got it,  guessing for perf. Please leave as is then